tasks/agent.py

```diff
@@ -651,7 +651,7 @@ def check_supports_python_version(_, check_dir, python):
         else:
             print('False', end='')
     elif os.path.isfile(setup_file):
-        with open(setup_file, 'r') as f:
```

Comment:
Hello, why remove read only mode on this ? (mainly asked for curiosity)

Reply:
That's a good question! This is the default value on Python 3, so there's no need to specify it and ruff auto-fixes it

Reply:
The default mode is already read-only (cf. https://docs.python.org/3/library/functions.html#open) and `ruff` has this rule: https://docs.astral.sh/ruff/rules/redundant-open-modes/

Reply:
okay thanks a lot